napari/_qt/layer_controls/qt_labels_controls.py

```diff
@@ -454,6 +456,11 @@ def _on_editable_change(self, event=None):
             self.layer.editable,
         )
 
+    def close(self):
+        """Layer widget is closing."""
+        self.colorBox.close()
```

Comment:
just a note here: parents close all of their children _by default_ (this is why it's important to instantiate QWidgets with a parent, or assign one later)... This _may_ not be necessary at all, but it may have been added to pass the pytest "leaked widgets" tests.  If we remove it and the leaked widget test passes, it wasn't necessary.

Reply:
I think I needed to add it to make the "leaked widgets" test pass, but I will confirm before merge

Reply:
Ah no - this was actually needed because `colorBox` has layer event connections which must be disconnected. I've added a comment

Reply:
Sure, but the question is: who is the colorBox's parent if not `self` here, and if it's self, won't `self.colorBox.close` be called automatically as @tlambert03 says?

Reply:
ok, now looking through children that have a close method and calling that in the base widget